Sources/JExtractSwift/FFM/FFMSwift2JavaGenerator.swift

```diff
@@ -16,27 +16,87 @@ import JavaTypes
 import SwiftSyntax
 import SwiftSyntaxBuilder
 
+package class FFMSwift2JavaGenerator: Swift2JavaGenerator {
+  let log = Logger(label: "ffm-generator", logLevel: .info)
+
+  let analysis: AnalysisResult
+  let swiftModuleName: String
+  let javaPackage: String
+  let swiftOutputDirectory: String
+  let javaOutputDirectory: String
+  let swiftStdlibTypes: SwiftStandardLibraryTypes
+  let symbolTable: SwiftSymbolTable
+
+  var javaPackagePath: String {
+    javaPackage.replacingOccurrences(of: ".", with: "/")
+  }
+
+  var thunkNameRegistry: ThunkNameRegistry = ThunkNameRegistry()
+
+  /// Cached Java translation result. 'nil' indicates failed translation.
+  var translatedSignatures: [ImportedFunc: TranslatedFunctionSignature?] = [:]
+
+  package init(
+    translator: Swift2JavaTranslator,
+    javaPackage: String,
+    swiftOutputDirectory: String,
+    javaOutputDirectory: String,
+  ) {
+    self.analysis = translator.result
+    self.swiftModuleName = translator.swiftModuleName
+    self.javaPackage = javaPackage
+    self.swiftOutputDirectory = swiftOutputDirectory
+    self.javaOutputDirectory = javaOutputDirectory
+    self.symbolTable = translator.symbolTable
+    self.swiftStdlibTypes = translator.swiftStdlibTypes
+  }
+
+  func generate() throws {
+    try writeExportedJavaSources()
+    try writeSwiftThunkSources()
```

Comment:
Just curious, any reason you changed the order? The current main writes the Swift thunk files first.

Reply:
hm no, I can change it back if you'd prefer that?